---
title: "Get Data"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{Get Data}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---
```{r setup, include = FALSE}
knitr::opts_chunk$set(warning = FALSE, fig.width = 7, fig.height = 5, fig.align = "center")
```

```{r load_packages, message = FALSE, warning = FALSE}
#Load packages
library(GAMBLR.data)
library(dplyr)
```

# Get Data
This vignette explores how to retrieve different data types bundled withing GAMBLR.data. Commonly, such GAMBLR functions are prefixed with `get_`. These functions are readily available for returning data of different types. Such as Simple Somatic Mutations (SSM), Copy Number (CN) segments and Structural Variants (SV). This resource explores commonly occurring parameters across different functions, best-practices and recommendations in the scope of retrieving GAMBLR data. This vignette is built using the bundled data in GAMBLR.data and thus, the dimensions reported here are in respect to what is available in the bundled data. Note, in all examples throughout this vignette, sample IDs are being used that are readily available in both setups. 

## Metadata
First, lets start with retrieving metadata for all GAMBL samples with `seqtype` *genome*. To return metadata for capture samples, set `seq_type_filter = "capture"`. It is also possible to return metadata for more than one seq type, e.g `seq_type_filter = c("genome", "capture")`. Depending on what GAMBLR setup you are running (GAMBLR.results or GAMBLR.data) the return will be different. While we are at it, let us also printout the column names (metadata information) we have at our disposal.
```{r metadata}
#Get gambl metadata for all samples
this_metadata = get_gambl_metadata(seq_type_filter = "genome")

#The dimensions of the returned metadata object
dim(this_metadata)

#These are the metadata columns that are available within the bundled data
colnames(this_metadata)

#Subset metadata on a set of samples (samples classified as DLBCL for pathology)
dlbcl_metadata = dplyr::filter(this_metadata, 
                               pathology == "DLBCL")

#Dimension of metadata subset (restricted to samples with pathology classified as DLBCL)
dim(dlbcl_metadata)
```

## Get SSM
Based on the information available to the user, application, or downstream analysis, there are multiple ways to retrieve SSM data. For example, if the user knows the sample ID and is only interested in looking at SSM results for that particular sample, one can use `get_ssm_by_sample`. If multiple samples are to be queried, `get_ssm_by_samples` is recommended. The user can also use patient IDs for retrieving this data, in this case `get_ssm_by_patients` is available. In addition, the user can also restrict SSM calls to specific genomic regions with `get_ssm_by_regions` or `get_ssm_by_region`. Another possibility for returning coding SSM is to call `get_coding_ssm`, this function returns all coding SSMs from the GAMBL database in MAF-like format. If the user have an already subset metadata, the `these_samples_metadata` parameter can be used with all SSM functions to restrict the variants returned to the sample IDs in this data frame, handy! In this section we will be demonstrating how these functions can be used with a variety of parameters and parameter combinations.

### By Samples
Return SSMs for one or more samples with `get_ssm_by_samples`. In the examples bellow, we are requesting SSM for the DOHH-2 cell line in two different ways, first using the `these_sample_ids` parameter and next with giving the function a metadata table that has already been subset to the sample ID of interest. In the third example, we are further restricting the returned SSM to mutations withing a specified gene **MYC** with the parameter `these_genes` and only keeping variants with a minimum read support of 3. Lastly, it is also possible top dictate what MAF columns we want back. To do this, call the function with the `maf_col` parameter and specify the columns of interest as a vector of characters. In example 5 we are requesting SSM calls for multiple samples (using all sample IDs available in the bundled metadata). Note, if we do not give this function any sample IDs or metadata to work with, the function will per default return SSM for all samples available in the bundled metadata.

#### Example 1
Using the `these_sample_id` parameter. Other parameters are left at default in this example. Let us also print the dimensions of the return MAF (i.e how many mutations do we get back and the names of the available MAF columns.
```{r get_ssm_by_samples_example_1}
#Example 1 - Using the these_samples_id parameter
sample_dohh2_ssm = get_ssm_by_samples(these_sample_ids = "DOHH-2")

#How manny mutations do we get back?
dim(sample_dohh2_ssm)

#What columns are available?
colnames(sample_dohh2_ssm)

#What Variant Types are called in this MAF?
table(sample_dohh2_ssm$Variant_Type)

#What are the available Variant Classifications?
table(sample_dohh2_ssm$Variant_Classification)
```
#### Example 2
Requesting the same sample ID but this time using a metadata table that has already been subset to the sample ID of interest. We can also confirm that this method creates the same return as it would if we used `these_sample_ids` as in the previous example. For this reason, let us also print the dimensions of the returned MAF and compare it with the previous example.
```{r get_ssm_by_samples_example_2}
#Subset a metadata to the sample ID of interest (DOHH-2)
dohh2_meta = get_gambl_metadata(seq_type_filter = "genome") %>% dplyr::filter(sample_id == "DOHH-2")

#Run get_ssm_by_sample with metadata dictating what samples are of interest
meta_dohh2_ssm = get_ssm_by_samples(these_samples_metadata = dohh2_meta)

#How many mutations do we get back?
dim(meta_dohh2_ssm)

#Is the function returning the same data in the two examples?
all.equal(meta_dohh2_ssm, sample_dohh2_ssm)
```

#### Example 3
Let us explore some of the other parameters that are available. Here we are requesting the same sample ID (DOHH-2) but in respect to another genome build (hg38). In addition, the returned mutation calls are limited to the ones intersecting with MYC and that have a read support of at least 3 reads. Lastly, we are also calling this function with `verbose = TRUE`. 
```{r get_ssm_by_samples_example_3}
#Example 3 - Use a combination of different parameters provided by this function
this_ssm = get_ssm_by_samples(these_sample_ids = "DOHH-2",
                              projection = "hg38",
                              these_genes = "MYC",
                              min_read_support = 3,
                              verbose = TRUE)

#Print the returned MAF, subset to the first 10 columns
select(this_ssm, 1:10)
```

#### Example 4
In this last example, we are requesting SSM calls for all capture samples that are available in the bundled metadata. We are also setting `basic_columns = FALSE` which returns two additional columns. Note, in the GAMBLR.results version of this function, setting basic columns to FALSE will return a MAF with 116 columns. This is because not all MAF columns are kept in the bundled version of this data set.
```{r get_ssm_by_samples_example_4}
#Get capture metadata
capture_meta = get_gambl_metadata(seq_type_filter = "capture")

#Get capture ssm
capture_ssm = get_ssm_by_samples(these_samples_metadata = capture_meta,
                                 this_seq_type = "capture",
                                 basic_columns = FALSE)

#What are the dimensions for the capture SSM calls?
dim(capture_ssm)

#For how many capture samples do we have SSM calls for?
length(unique(capture_ssm$Tumor_Sample_Barcode))

#What are the extra columns we get when basic_columns = FALSE?
colnames(select(capture_ssm, 46:47))
```
#### Example 5
It is also possible to query a specify set of sample IDs of interest with the `these_sample_ids` parameter, as such.
```{r get_ssm_by_samples_example_5}
these_samples_ssm = get_ssm_by_samples(these_sample_ids = c("SU-DHL-4",
                                                            "DOHH-2",
                                                            "SU-DHL-10"))
```

#### Example 6
If no sample IDs or metadata is provided, the function will per default return all SSM calls in the specified seq type. Let us also call this example using the `maf_cols` parameter to control what columns are returned. Note, that if we want to dictate what columns we want back, we need to call the function with `basic_columns = FALSE` otherwise, the function will disregard anything specified under `maf_cols`.
```{r get_ssm_by_samples_example_6}
all_ssm = get_ssm_by_samples(this_seq_type = c("genome", "capture"), 
                             basic_columns = FALSE, 
                             maf_cols = c("Tumor_Sample_Barcode",
                                          "Variant_Type",
                                          "t_alt_count",
                                          "Variant_Classification"))

#In total, how many samples do we have SSM calls for?
length(unique(all_ssm$Tumor_Sample_Barcode))

#Can we control what columns we get back (as advertised)
colnames(all_ssm)
```

### By Patient
Return SSM for a set of patients with `get_ssm_by_patients`. Similarly to `get_ssm_by_samples` this function can accept the same parameters, allowing the user more control of what gets returned. This is because `get_ssm_by_patients` internally calls `get_ssm_by_samples` but with an additional data processing step that takes a patient ID and converts it to a sample ID (which is forwarded to `get_ssm_by_samples`). Note that in the example bellow, the patient ID and sample ID are the same, this is not true for all samples.
```{r get_ssm_by_patients}
#get_ssm_by_patients
these_patients_ssm = get_ssm_by_patients(these_patient_ids = "DOHH-2")

#Do we get the same MAF back as we do with get_ssm_by_samples, with the same sample specified?
all.equal(these_patients_ssm, sample_dohh2_ssm)
```

### By Region
In this section we are exploring the different way you can call `get_ssm_by_region` and `get_ssm_by_regions`.

#### Example 1
Return SSM based on a specific region with `get_ssm_by_region`. In this example we are using `gene_to_region` from GAMBLR.utils to get genomic coordinates for a specific gene that will be called by `get_ssm_by_region`.
```{r get_ssm_by_region_example_1}
#Get the region back for MYC (grch37)
myc_region = GAMBLR.utils::gene_to_region(gene_symbol = "MYC", 
                                          return_as = "region")

#Get all SSMs that are residing in the defined MYC region
myc_ssm = get_ssm_by_region(region = myc_region)

#Do we get the same mutations back as if we're calling get_ssm_by_samples with these_genes restricted to MYC?
all.equal(select(myc_ssm, 1:45), get_ssm_by_samples(these_genes = "MYC"))
```

#### Example 2
We can also specify the coordinates we are interested in using the `chromosome`, `qstart`, and `qend` parameters, Let us also check if the returned MAF is the same we get back with the `region` parameter.
```{r get_ssm_by_region_example_2}
myc_ssm_chrom = get_ssm_by_region(chromosome = 8,
                                  qstart = 128747680,
                                  qend = 128753674)

#Comapre the return, are they the same?
all.equal(myc_ssm, myc_ssm_chrom)
```

#### Example 3
It is also possible to call this function and provide an already loaded MAF object to subset to the desired regions. To do so, specify the MAF of interest with `maf_data`. 
```{r get_ssm_by_region_example_3}
dohh2_myc_ssm = get_ssm_by_region(maf_data = sample_dohh2_ssm,
                                  chromosome = 8,
                                  qstart = 128747680,
                                  qend = 128753674)

```

#### Example 4
In addition to the previous explored get_ssm function, this function also has a streamlined option available. If set to TRUE, only two columns are kept in the returned MAF (start_position and Tumour_Sample_Barcode). In this example, we are extending the coordinates to encapsulate the whole p-arm of chromosome 1. Note, since we are requesting hg38 calls, we have to add the chr prefix for the specified chromosome. Let us also further restrict the returned mutation calls to the ones with a read support of at least five.
```{r get_ssm_by_region_example_4}
streamlined_ssm = get_ssm_by_region(chromosome = "chr1",
                                    qstart = 10000,
                                    qend = 121700000, 
                                    streamlined = TRUE, 
                                    projection = "hg38", 
                                    min_read_support = 5)

#What columns are kept with streamlined = TRUE?
colnames(streamlined_ssm)

#How many variants do we find within the specified region?
nrow(streamlined_ssm)
```

#### Example 5
Similarly to other get_ssm functions we can also call `get_ssm_by_region` with either `these_sample_ids` or `these_samples_metadata` to restrict the return to specific sample IDs. In this example, we a re first calling the function with `these_sample_ids = "DOHH-2"` and then with `these_samples_metadata` giving the function a metadata that has been subset to the same sample ID. Note, The metadata table given to the function can obviously contain more than one sample ID and the function will return SSM calls for all such samples IDs, given that they have SSM calls in the bundled data.
```{r get_ssm_by_region_example_5}
dohh2_region_sample = get_ssm_by_region(these_sample_ids = "DOHH-2", 
                                        region = myc_region)

dohh2_region_metadata = get_ssm_by_region(these_samples_metadata = dohh2_meta,
                                          region = myc_region)

#Are the two returned data frames identical?
all.equal(dohh2_region_sample, dohh2_region_metadata)
```

#### Example 6
For multiple regions, refer to `get_ssm_by_regions`. In this example we are returning SSM calls for all ASHM regions (hyper mutated) across all available samples. With this multiple-region-version of this function we also get the region name added to the returned data frame and there are a couple of different ways this can be done. If you are providing regions as a bed file (`regions_bed`), you have the option of setting `use_name_column = TRUE`. If you do so your bed file can have a fourth column simply named "name", in this case, the function will keep this column for naming the returned regions in the MAF. With `streamlined = TRUE` the function returns the minimal number of columns.
```{r get_sm_by_regions_example_6}
#Get aSHM genes, select the columns of interest and rename for get_ssm_by_regions compatibility 
ashm_regions = grch37_ashm_regions %>% 
  dplyr::select(chr_name, hg19_start, hg19_end, gene) %>% 
  dplyr::rename(chromosome = chr_name, start = hg19_start, end = hg19_end, name = gene)

#Get ssm for all ashm regions
ashm_ssm = get_ssm_by_regions(regions_bed = ashm_regions, 
                              use_name_column = TRUE, 
                              streamlined = TRUE)

head(ashm_ssm)
```

#### Example 7
If the use instead wish to specify regions as a vector of characters (`regions_list`) instead of using a bed file. The function will not accept a fourth element for naming the returned regions. If so, the function defaults to printing the specified regions instead.
```{r get_sm_by_regions_7}
ssm_region_list = get_ssm_by_regions(regions_list = c("chr1:6661482-6662702",
                                                      "chr2:232572640-232574297",
                                                      "chr17:75443766-75451177"), 
                                     streamlined = TRUE)

head(ssm_region_list)
```

#### Example 8
In the last example of this function, let us use a combination of non-default parameters, just for fun! Please note, that the specified regions must be formatted to match the specified projection. This means if `projection = "hg38"` chromosome names must be "chr" prefixed. If `projection = "grch37"` then chromosome names must not be prefixed.
```{r get_sm_by_regions_8}
#Create a data frame in bed format
my_regions = data.frame(chr = c("chr1", "chr2", "chr17"), 
                        start = c("10000", "10000", "10000"), 
                        end = c("248956422", "242193529", "83257441"),
                        name = c("Chromosome 1", "Chromosome 2", "Chromosome 17"))

fun_ssm = get_ssm_by_regions(these_sample_ids = "Akata",
                             projection = "hg38",
                             regions_bed = my_regions,
                             streamlined = TRUE,
                             use_name_column = TRUE,
                             min_read_support = 1,
                             verbose = TRUE)
```

### Coding SSM
Lastly, another way to retrieve SSM is to call `get_coding_ssm`. This function returns coding SSM for any given sample. This function is a convenient option for anyone interested in plotting coding mutations. Convenient filtering parameters are included in this function for easy and straight-forward subset to specific pathologist, cohorts, and sample IDs. If these parameters are not called, coding SSM will be returned for all samples. In the example bellow, we are calling this function with no samples or metadata provided, instead we're using the in-0house filtering options to subset the return to our desire. Of course, the user can provide a metadata subset that has already been filtered to the sample IDs of interest (using `these_samples_metadata`).
```{r get_coding_ssm}
#First, let us have a look at all the valid cohorts and pathology
unique(get_gambl_metadata()[,"cohort"])
unique(get_gambl_metadata()[,"pathology"])

#Limit_cohort
dlbcl_cell_lines = get_coding_ssm(these_samples_metadata = get_gambl_metadata(),
                                  limit_cohort = "DLBCL_cell_lines")
dim(dlbcl_cell_lines)

#Exclude_cohort
no_dlbcl_cell_lines = get_coding_ssm(exclude_cohort =  "DLBCL_cell_lines")
dim(no_dlbcl_cell_lines)

#limit_pathology
fl_coding_ssm = get_coding_ssm(limit_pathology = "FL")
dim(fl_coding_ssm)

#exclude silent mutations
no_silent = get_coding_ssm(these_sample_ids = "DOHH-2",
                           include_silent = FALSE)

#Do we have any variants classified as "silent"?
table(no_silent$Variant_Classification)
```

## Get CN
For the purpose of retrieving CN data, we have two functions available `get_sample_cn_segments` and `get_sn_segments`. Each with its own specialized application and recommended usage. Briefly, `get_sample_cn_segments` is best called if the user wants to query CN segments for a specific subset set of samples. This function works for singular as well as multiple samples. In addition, many parameters and their behavior should be familiar from the previous section (such as `this_seq_type`, `projection`, `these_sample_ids`, `these_samples_metadata`, and `streamlined`). If the user instead wants to query CN calls for a specific region or genomic loci, `get_cn_segments` is best used. In this section we will explore the two different functions and demonstrate how they can be used.

### CN By Samples
`get_sample_cn_segments` Returns CN segments for single sample or multiple samples. Specify the sample IDs you are interested in with `these_sample_ids` (as a vector of characters), or call this function with `these_samples_metadata` if you already have a metadata table subset to the sample IDs of interest. If none of the above parameters are specified, the function will return CN segments for available samples (from `get_gambl_metadata`).

#### Example 1
For starters, let us call this function with default parameters. This returns all available CN information from the bundled data. This is because, under the hood this function calls `id_ease` that is extracting sample IDs from the metadata for all available samples (if not `these_sample_ids` or `these_samples_metadata` is provided). Note, that default seq type is genome, this is because there are currently no capture samples with CN data available in the bundled data.
```{r get_sample_cn_segments_example_1}
all_segs = get_sample_cn_segments()

#What are the dimensions of the returned object?
dim(all_segs)

#What are the columns we have available?
colnames(all_segs)
```

#### Example 2
Now, let us explore some of the other parameters we have available. In this example we are calling the function with `these_sample_ids = "DOHH-2"` to return CN data for. We are also using a metadata subset restricted to the same sample ID to demonstrate that either of these parameters will return the same data, as long as they have the same sample ID.
```{r get_sample_cn_segments_example_2}
#Use these_sample_ids
sample_dohh2_segs = get_sample_cn_segments(these_sample_ids = "DOHH-2")

#Use these_samples_metadata
meta_dohh2_segs = get_sample_cn_segments(these_samples_metadata = dohh2_meta)

#Are they the same?
all.equal(sample_dohh2_segs, meta_dohh2_segs)

#Let us also have a look at how many CN segments are returned for this sample, as well as them reported CN states
nrow(sample_dohh2_segs)
table(sample_dohh2_segs$CN)
```

#### Example 3
In this example, let us retrieve CN segments for multiple samples, again utilizing the two parameters `these_sample_ids` and `these_samples_metadata`. To mix it up, we can also request a different projection.
```{r get_sample_cn_segments_example_3}
#First, we can create a metadata subset with the samples of interest. Here we are getting all samples form the DLBCL_cell_line cohort
dlbcl_metadata_cellline = get_gambl_metadata() %>% 
  dplyr::filter(cohort == "DLBCL_cell_lines")

#Next, let us pull the sample IDs from this metadata subset
dlbcl_samples_cellline = dlbcl_metadata_cellline$sample_id

#What are the available sample IDs from this cell line?
dlbcl_samples_cellline

#Call get_sample_cn_segments
dlbcl_samples_segs = get_sample_cn_segments(these_sample_ids = dlbcl_samples_cellline, 
                                            projection = "hg38")

dlbcl_meta_segs = get_sample_cn_segments(these_samples_metadata = dlbcl_metadata_cellline, 
                                         projection = "hg38")

#Are they the same?
all.equal(dlbcl_samples_segs, dlbcl_meta_segs)

#We can also look at the sample IDs returned in this seg file to see what samples from the selected cell line we do have CN data for
unique(dlbcl_samples_segs$ID)
```

#### Example 4
In this last example we will demonstrate how the other parameters can be used, providing even more control of the return from this function. `stremalined = TRUE` restricts the columns to only return sample ID (ID) and the absolute copy number state (CN). `with_ch_prefix` controls if the chromosomes should be prefixed with **chr** or not (default is FALSE). Lastly, we have a verbose option that can be useful for debugging purposes.
```{r get_sample_cn_segments_example_4}
#Get segments with chr prefix (even though grch37 is requested, as the default projection)
my_prefixed_segs = get_sample_cn_segments(with_chr_prefix = TRUE,
                                          verbose = TRUE)

#print the first five rows of the returned data frame. Note chromosomes are prefixed in thsi return.
head(my_prefixed_segs)

#return the streamlined option
minimal_seg = get_sample_cn_segments(streamlined = TRUE)

#print first five rows, note we only get two columns back in the data frame
head(minimal_seg)
```

### Get CN Within A Genomic Range
`get_scn_segments` behaves very similar to `get_sample_cn_segments` with the difference that the user can specify a genomic loci to restrict the returned CN segments to. In order to achieve this, the function have the following additional parameters available; `region`, `chromosome`, `qstart`, and `qend`. The user can provide the full region in a "region" format (chr:start-end) to the `region` parameter. Or, the user can provide chromosome, start and end coordinates individually with `chr`, `qstart`, and `qend` parameters. In the following examples we will demonstrate how this can be achieved in a few different ways.

#### Example 1
Here we are showing how to give the function a complete region with the `region` parameter. Note, we can integrate useful utility functions from GAMBLR.utils to get regions back in the correct format for any given gene. Similarly to `get_sample_cn_segmetns` if `these_sample_ids` and/or `these_samples_metadata` are not specified, the function will query all available samples. For demonstration purposes, let us also call this example on the DOHH-2 metadata subset.
```{r get_cn_segments_example_1}
#Get the regions for BCL2
bcl2_region = GAMBLR.utils::gene_to_region(gene_symbol = "BCL2", 
                                           genome_build = "hg38")

#Print this region
bcl2_region

#Get cn segments in this region for all samples
bcl2_segs = get_cn_segments(region = bcl2_region,
                            projection = "hg38",
                            with_chr_prefix = TRUE)

#What are the dimensions of the returned data frame?
dim(bcl2_segs)

#How many samples have reported CN variation in the queried region?
length(unique(bcl2_segs$ID))

#What are the CN states in this region for the selected samples?
table(bcl2_segs$CN)

#Let us call this function using the DOHH-2 metadata subset as well
dohh2_bcl2_segs = get_cn_segments(these_samples_metadata = dohh2_meta, 
                                  region = bcl2_region)

#How many reported CN states are within BCL2 for DOHH-2?
nrow(dohh2_bcl2_segs)

#Do we get CN states back for DOHH-2 as the only sample?
unique(dohh2_bcl2_segs$ID)
```

#### Example 2
In this example we are instead specifying the region of interest with the `chromosome`, `qstart`, and `qend` parameters. As a sanity check we are also comparing the returned data frame with what we get when specifying the same region with the `region` parameter.
```{r get_cn_segments_example_2}
#Let us call a helper function for converting the BCL2 region into chunks
bcl2_chunks = region_to_chunks(region = bcl2_region)

#What does the chunks look like?
bcl2_chunks

bcl2_chunks_segs = get_cn_segments(chromosome = bcl2_chunks$chromosome, 
                                   qstart = bcl2_chunks$start, 
                                   qend = bcl2_chunks$end, 
                                   projection = "hg38", 
                                   with_chr_prefix = TRUE)

#Does this return match what we get when specifying the same genomic range with `range` in the previous example?
all.equal(bcl2_segs, bcl2_chunks_segs)
```

### CN and SSM
There is also a specialized function available called `assing_cn_to_ssm` that takes a sample ID with the `this_sample_id` parameter and annotates mutations with copy number information. A variety of parameters are at hand for a customized workflow. For example, the user can specify if only coding mutations are of interest. To do so, set `coding_only = TRUE`. This function internally calls `get_ssm_by_samples` and `get_sample_cn_segments`. This function can also take a vector with genes of interest (`genes`) that the returned data frame will be restricted to. The return for this function is a list. This list contains a maf (returned with `get_ssm_by_sample`) with three extra columns; LOH, log.ratio and CN that annotates the absolute CN state within that SSM call.

#### Example 1
In this example we are calling `assign_cn_to_ssm` on DOHH-2 with default parameters.
```{r assign_cn_to_ssm_example_1}
dohh2_ssm_cn = assign_cn_to_ssm(this_sample_id = "DOHH-2", 
                                projection = "hg38", 
                                genes = "MYC",  
                                coding_only = TRUE)
#What is the return?
typeof(dohh2_ssm_cn)

#What does the elements in the list look like?
head(dohh2_ssm_cn$maf)
head(dohh2_ssm_cn$seg)
```

### Get Manta SV
In this last section we will explore hoe to get Manta SV data using GAMBLR.data. For this purpose `get_manta_sv` was developed. This function has a collection of (by now) familiar parameters. For example, To obtain SV calls for multiple samples, give `these_sample_ids` a vector of sample IDs. Alternatively, the user can also provide the `these_samples_metadata` parameter to make use of an already subset metadata table. In this case, the returned SVs will be restricted to the sample_ids within that data frame. This function can also restrict the returned calls to any genomic regions specified within `chromosome`, `qstart`, `qend`, or the complete region specified under `region` (in chr:start-end format), note that chromosome can be either prefixed or not prefixed. In addition, useful filtering parameters are also available, use `min_vaf` to set the minimum tumour VAF for a SV to be returned and `min_score` to set the lowest Manta somatic score for a SV to be returned. `pair_status` can be used to return variants from either matched or unmatched samples. In addition, the user can chose to return all variants, even the ones not passing the filter criteria. To do so, set `pass = FALSE` (default is TRUE).

#### Example 1
In this first example we will call `get_manta_sv` with default parameters to see what we get back.
```{r get_manta_sv_example_1}
#Default parameters
all_manta = get_manta_sv()

#How many SVs do we get back?
nrow(all_manta)

#How many samples do we have SV calls for?
length(unique(all_manta$tumour_sample_id))

#What does the returned data frame looks like?
head(all_manta)
```

#### Example 2
Here we are demonstrating the sample ID subset options
```{r get_manta_sv_example_2}
#Use these_sample_ids
sample_dohh2_manta = get_manta_sv(these_sample_ids = "DOHH-2")

#Use these_samples_metadata
meta_dohh2_manta = get_manta_sv(these_samples_metadata = dohh2_meta)

#Are they the same?
all.equal(sample_dohh2_manta, meta_dohh2_manta)

#How many SV calls do we get back for the selected sample?
nrow(sample_dohh2_manta)

#Where in the genome are these SV calls?
dplyr::select(sample_dohh2_manta, CHROM_A, START_A, END_A, CHROM_B, START_B, END_B)
```

#### Example 3
From the previous example, it looks like the reported SVs are residing within MYC. Let us call `get_manta_sv` specifying this region first in the region-format  and then with specifying the chromosome, start and end individually. Let us also use less stringent VCF filters, to get as much back as possible.
```{r get_manta_sv_example_3}
#Specifying MYC in region format
dohh2_myc_manta_region = get_manta_sv(these_sample_ids = "DOHH-2", 
                                      region = myc_region, 
                                      min_vaf = 0, 
                                      min_score = 0, 
                                      pass = FALSE)

#Specifying MYC with chromosome, qstart and qend parameters
dohh2_myc_manta_chunks = get_manta_sv(these_sample_ids = "DOHH-2", 
                                      chromosome = 8, 
                                      qstart = 128747680, 
                                      qend = 128753674, 
                                      min_vaf = 0, 
                                      min_score = 0, 
                                      pass = FALSE)

#Are the returns data frames the same?
all.equal(dohh2_myc_manta_region, dohh2_myc_manta_chunks)
```

#### Example 4
Here we are demonstrating the VCF filtering options. In this example we are calling `get_manta_sv` on the DLBCL metadata subset. For demonstration purposes, we are also requesting a non-default projection and setting `verbose = TRUE` for more information printed to the console.
```{r get_manta_sv_example_4}
#Get manta SVs for the DLBCL cell line
dlbcl_manta = get_manta_sv(these_samples_metadata = dlbcl_metadata_cellline,
                           projection = "hg38",
                           verbose = TRUE,
                           min_vaf = 0.4, 
                           min_score = 100)

#How many variants do we get back with these filters?
nrow(dlbcl_manta)

#Do the advertised VCF filters work?
all(dlbcl_manta$VAF_tumour >= 0.4)

#Do the advertised VCF filters work?
all(dlbcl_manta$SCORE >= 100)
```
